### DISCUSSION TITLE
Support multiple states in state trigger to/from fields

### DISCUSSION THREAD

**Discussion by wharwood** [home-assistant/frontend#17875] [Automations / scripts / scenes]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

In the automation trigger visual editor, supplying a comma separated list of conditions in the "to:" and "from:" fields results in the automation checking for a monolithic state that will always resolve to false. For example, if one supplies the following list of states `A,B,C,D` the resulting YAML resolves as `to: A,B,C,D`. Similarly, if one supplies `'A','B','C','D'`, `"A","B","C","D"`, `A, B, C, D`, `'A', 'B', 'C', 'D'`, or `"A", "B", "C", "D"` all resolve to the monolithic, single line result like the aforementioned case.

### Describe the behavior you expected

I would expect the conditions to be separated and evaluated individually. If one were to supply `A,B,C,D`, the corresponding YAML would produce the following. 

```
to:
   - A
   - B
   - C
   - D
```

Conversely, a UI element that allows the addition of multiple to and from states to be added in the same vein as trigger or conditions themselves would likely be more clear to the end user.

### Steps to reproduce the issue

1. Set up an automation triggered by an entity with multiple string states (like a weather condition)
2. Place a comma separated list of conditions in the to or from fields of the trigger (like `rainy, snowy`)
3. Enable the automation
4. Change the entity state to or from the states used (dependent on the field and states used in step 2)
5. Verify that the trigger did not fire
6. Switch the trigger mode from Visual to YAML and confirm that the to or from field in a monolithic state instead of the multiple states intended
...

### What version of Home Assistant Core has the issue?

Home Assistant 2023.8.3

### In which browser are you experiencing the issue with?

Google Chrome Version 115.0.5790.170 (Official Build) (arm64)

### Which operating system are you using to run this browser?

macOS Ventura 13.2.1 (22D68)

**Comment by tdejneka** [comment]
> If one were to supply A,B,C,D, the corresponding YAML would produce the following.

The `to/from` options require a **list** value. 

`A, B, C, D` is a comma-delimited **string**.

There's no bug here; if you don't supply a list, it's handled as a string.

---

You can submit a Feature Request but be advised that automatically converting a comma-delimited string into a list would thwart the ability to specify an actual comma-delimited string value.

**Comment by karwosts** [comment]
If this was implemented we could have additional combo boxes spawn each time you select a state so that multiple could be selected, similar to how you can add multiple entities. 

That said not certain if that's a desired feature or not, I know the goal is not to implement every single functionality of the yaml into the UI, but just what is most simple and unconfusing. 

As this is a feature request and not a bug (design is currently working as intended), I will move this to discussion area.

**Reply by The00Dustin** [reply]
Personally, I don't like it when I'm in the visual editor and it says "Visual editor is not supported for this configuration" about simple things (I get why templates and notifications with variables and things like that fall back to YAML).  With scene triggers, I have multiple automations with state triggers that have more than one state, so what you suggest (as opposed to breaking commas in fields) seems like a good improvement to me.
